### ISSUE TITLE
Error while loading finetuned wav2vec 2.0 large

### ISSUE THREAD

**Issue by Sreyan88** [s3prl/s3prl#64]
Hi,
As per the ppt I try to load wav2vec2 with the following code and get the following error:

>>> upstream = torch.hub.load("s3prl/s3prl",'wav2vec2_url',ckpt = 'https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_vox_new.pt')
Using cache found in /home/sreyan/.cache/torch/hub/s3prl_s3prl_master
Using cache found in /home/sreyan/.cache/torch/hub/s3prl_cache/1c76d6e88090f01736036b28dc995fef583f47f42662d55286332557f957609f
for https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_vox_new.pt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sreyan/.conda/envs/semeval/lib/python3.7/site-packages/torch/hub.py", line 370, in load
    model = _load_local(repo_or_dir, model, *args, **kwargs)
  File "/home/sreyan/.conda/envs/semeval/lib/python3.7/site-packages/torch/hub.py", line 399, in _load_local
    model = entry(*args, **kwargs)
  File "/home/sreyan/.cache/torch/hub/s3prl_s3prl_master/upstream/wav2vec2/hubconf.py", line 23, in wav2vec2_url
    return wav2vec2_local(_urls_to_filepaths(ckpt, refresh=refresh), *args, **kwargs)
  File "/home/sreyan/.cache/torch/hub/s3prl_s3prl_master/upstream/wav2vec2/hubconf.py", line 14, in wav2vec2_local
    return _UpstreamExpert(ckpt, *args, **kwargs)
  File "/home/sreyan/.cache/torch/hub/s3prl_s3prl_master/upstream/wav2vec2/expert.py", line 24, in __init__
    model, cfg, task = fairseq.checkpoint_utils.load_model_ensemble_and_task([ckpt])
  File "/home/sreyan/fairseq/fairseq/checkpoint_utils.py", line 339, in load_model_ensemble_and_task
    state = load_checkpoint_to_cpu(filename, arg_overrides)
  File "/home/sreyan/fairseq/fairseq/checkpoint_utils.py", line 273, in load_checkpoint_to_cpu
    state = _upgrade_state_dict(state)
  File "/home/sreyan/fairseq/fairseq/checkpoint_utils.py", line 550, in _upgrade_state_dict
    state["cfg"] = convert_namespace_to_omegaconf(state["args"])
  File "/home/sreyan/fairseq/fairseq/dataclass/utils.py", line 351, in convert_namespace_to_omegaconf
    with initialize(config_path=config_path):
AttributeError: __enter__

I would want to finetune finetuned wav2vec 2.0 on speech sentiment task. Any help would be highly appreciated.

**Comment by andi611** [comment]
Hi,
This looks like a fairseq error, do you have the latest version of fairseq?
You can upgrade to 0.10.2 by: `pip install fairseq --upgrade`

Or change to another checkpoint other than: https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_vox_new.pt.
We've tested these two:
wav2vec 2.0 - large: https://dl.fbaipublicfiles.com/fairseq/wav2vec/libri960_big.pt
wav2vec 2.0 - base: https://dl.fbaipublicfiles.com/fairseq/wav2vec/wav2vec_small.pt

**Comment by Sreyan88** [comment]
Hi Andi,
Yes my fairseq is latest version.

```
>>> import fairseq
>>> print(fairseq.__version__)
1.0.0a0+1164a7f
```

I receive the same error on 
wav2vec 2.0 - large:https://dl.fbaipublicfiles.com/fairseq/wav2vec/libri960_big.pt.

**Comment by leo19941227** [comment]
Hi!
I am happy to help. Since fairseq is still under quick development and some functions might not work on their latest `master`.
Could you tell me how did you install fairseq?
Here I use `pip install fairseq`, and my fairseq version is 0.10.2.

**Comment by Sreyan88** [comment]
Hi Leo,
Thanks a lot! That worked!

I would recommend adding audiomentations and gdown to requirements.txt. 

I had one more question. I am trying to use pre-trained wav2vec2 for speech sentiment classification downstream task, is that somehow possible using s3prl toolkit?

Thank You!

**Comment by leo19941227** [comment]
Hi Sreyan88,

Thanks for your recommendation! I will do that later.

Yes! You just need to add a self-contained folder specifying your downstream models and datasets. Unfortunately I still don't have much time to do the documentation, but maybe you can refer to the `downstream/`  folder. There are lots of downstream tasks we are currently implementing, and `downstream/example`/  provides a minimum requirement for adding a new downstream task. Maybe you can check the files in `downstream/example/`.

For a minimum run-up you can type
```shell
python3 run_downstream.py --mode train --upstream wav2vec2 --upstream_trainable --downstream example --expname HelloWorld
```

In a nutshell, you just need to implement the functions marked by `interface` in `downstream/example/expert.py`, and follow the yaml config style in `downstream/example/config.yaml` by only changing the `downstream_expert` field, which can by customized by you.

Your any suggestion is appreciated and the pull request is welcome once you finish the sentiment task! Thanks!
Note. We already support CMU-MOSEI sentiment analysis, in case that's the dataset you wish to use.

**Comment by Sreyan88** [comment]
Hi @leo19941227 ,

Thanks a lot for all the help. I will definitely check this out and let you know on how it goes!

**Comment by Sreyan88** [comment]
Also, can you please let me know how you have used the dataset for CMU-MOSEI experiments and github implementation? Have you used raw data or pre-processed with the help of CMU-Multimodal SDK?

**Comment by leo19941227** [comment]
Hi,
I will let my friend who implements this to help with this. While he might be a little bit busy these two days, if that is the case, we are sorry for the late and hope to answer your question soon.

**Comment by godiclili** [comment]
Hi @Sreyan88 , 
We used raw data from [http://immortal.multicomp.cs.cmu.edu/raw_datasets/](http://immortal.multicomp.cs.cmu.edu/raw_datasets/) 
We also used [CMU_MOSEI_Labels.csd in mmsdk](http://immortal.multicomp.cs.cmu.edu/CMU-MOSEI/labels/CMU_MOSEI_Labels.csd) to segment raw audio and get labels. 
The train / dev / test split follows [cmu_mosei_std_folds.py in mmsdk](https://github.com/A2Zadeh/CMU-MultimodalSDK/blob/master/mmsdk/mmdatasdk/dataset/standard_datasets/CMU_MOSEI/cmu_mosei_std_folds.py)
Files we used to preprocess MOSEI are stored in downstream/mosei/utility [link here](https://github.com/s3prl/s3prl/tree/master/downstream/mosei/utility)

**Comment by Sreyan88** [comment]
Hi @godiclili and @leo19941227 ,

Thank you so much for the help, I understand the usage now. However if raw dataset was used, as far as I infer from the Mockingjay paper, the model was pre-trained using Mel scaled spectrogram features, and I don't see the code in the mosei downstream folder that converts raw audio to Mel scaled spectrogram for the downstream sentiment task.

Please let me know if I am missing something here. I am considering that the same input features need to be used for downstream as used in upstream (pre-trained).

Thank You for your help.

**Comment by leo19941227** [comment]
Hi,

Actually, the feature extraction part is wrapped in the [upstream](https://github.com/s3prl/s3prl/tree/master/upstream/mockingjay) if you check the `PretrainedTransformer
` in that script. We do this to keep all pretrained models taking raw waveform as input without taking care of different preprocessing from different pretrained methods.

**Comment by Sreyan88** [comment]
Sure thank you got it! So I suppose I would need a mixture of code from upstream and downstream directories and make my own code to finetune Mockingjay on downstream sentiment task on your pre-trained model. Thank You for the help!

**Comment by andi611** [comment]
> Sure thank you got it! So I suppose I would need a mixture of code from upstream and downstream directories and make my own code to finetune Mockingjay on downstream sentiment task on your pre-trained model. Thank You for the help!

Hi there,

If you are trying to finetune on your own downstream task, please see the documentation [here](https://github.com/s3prl/s3prl/blob/master/upstream/README.md#use-upstream-models-in-your-own-project).
If you are trying to finetune on our downstream tasks, you can simply use the argument `--upstream_trainable`  (or equivalently `-f`) from `run_downstream.py`.

**Comment by Sreyan88** [comment]
Hi @andi611 and @leo19941227 ,

Thank you for all the help! I have gone through a major part of the code and mostly understood everything! I am building my own dataset for sentiment classification task and will be using pre-trained Mockingjay for finetuning on the downstream, Will update you guys soon on the progress and also create a pull request on my new dataset task if possible.

Thank You for the support!
